### PR TITLE
System check providers running

### PR DESCRIPTION
**Pull request by ArneTR**
Metric providers can be left behind on the system in case of an unclean shutdown.

This PR introduces a system check to look for running providers in different GMT instance and also adds a test case to test the guard clause to trigger.

### CODE REVIEW

File: metric_providers/base.py [hunk 1]
```diff
@@ -47,7 +47,17 @@ def __init__(
 
     # this is the default function that will be overridden in the children
     def check_system(self):
-        pass
+        cmd = ['pgrep', '-f', self._metric_provider_executable]
+        result = subprocess.run(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            check=False, encoding='UTF-8')
+        # returncode 1 sadly indicates program failure or no found process, so we need to go deeper
```
Comment (ribalba) [inline metric_providers/base.py:55]: The comment here could be a little clearer. 
```
EXIT STATUS
     The pgrep and pkill utilities return one of the following values upon exit:

     0       One or more processes were matched.

     1       No processes were matched.

     2       Invalid options were specified on the command line.

     3       An internal error occurred.
```
The code looks ok though

File: metric_providers/base.py [hunk 2]
```diff
@@ -47,7 +47,17 @@ def __init__(
...
+        if result.returncode not in [0, 1]:
+            raise subprocess.CalledProcessError(result.stderr, cmd)
+        if result.stdout != '':
```
Comment (ribalba) [inline metric_providers/base.py:58]: Couldn't you work with return codes here? If you are already looking at the code beforehand? The Empty string check is also ok though IMHO

**Review by ribalba — APPROVED** [review]

File: metric_providers/base.py (hunk 1 above)
Comment (ArneTR) [reply metric_providers/base.py:55]: I restructured it to make it clearer when reading the code

File: metric_providers/base.py (hunk 2 above)
Comment (ArneTR) [reply metric_providers/base.py:58]: see comment above. The restructuring makes the comment obsolte I think

**Pull request merged**